src/Schemes/AffineSchemes/Objects/Attributes.jl

```diff
@@ -0,0 +1,275 @@
+```
+"""
+@attr function dim(X::AbsSpec{<:Ring, <:MPolyQuoLocalizedRing})
+  return dim(saturated_ideal(modulus(OO(X))))
+end
+
+@attr function dim(X::AbsSpec{<:Ring, <:MPolyLocalizedRing})
+  # the following line is supposed to refer the problem to the
+  # algebra side of the problem
+  return dim(ideal(ambient_ring(X), [zero(ambient_ring(X))]))
+end
+
+@attr function dim(X::AbsSpec{<:Ring, <:MPolyRing})
+  return dim(ideal(ambient_ring(X), [zero(ambient_ring(X))]))
+end
+
+@attr function dim(X::AbsSpec{<:Ring, <:MPolyQuo})
+  return dim(modulus(OO(X)))
+end
+
+
+@Markdown.doc """
+    codim(X::Spec)
+
+In Oscar, we can compute for an affine scheme ``X``
+the ring ``R = ambient_ring(X)``. This allows to consider
+an embedding of ``X`` into ``Spec(R)``. This method returns
+the codimension of the image of this embedding in ``Spec(R)``.
+
+# Examples
+```jldoctest
+julia> X = affine_space(QQ,3)
+Spec of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
+
+julia> codim(X)
+0
+```
+"""
+@attr function codim(X::Spec)
+  return dim(ideal(ambient_ring(X), [zero(ambient_ring(X))])) - dim(X)
+end
+
+
+@doc Markdown.doc"""
+    name(X::Spec)
+
+Returns the current name of an affine scheme.
+This name can be specified via `set_name!`.
+
+# Examples
+```jldoctest
+julia> X = affine_space(QQ, 3)
+Spec of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
+
+julia> name(X)
+"unnamed affine variety"
+
+julia> set_name!(X, "affine 3-dimensional space")
+
+julia> name(X)
+"affine 3-dimensional space"
+```
+"""
+@attr String function name(X::Spec)
+  return "unnamed affine variety"
+end
+
+
+function set_name!(X::AbsSpec, name::String)
+  return set_attribute!(X, :name, name)
+end
+
+
+########################################################################
+# (2) Further attributes for AbsSpec
+########################################################################
+
+# TODO: Needed?
+
+@Markdown.doc """
+    defining_ideal(X::AbsSpec{<:Any, <:MPolyRing})
+
+This method return the defining ideal of an affine scheme ``X``.
+
+# Examples
+```jldoctest
+julia> X = affine_space(QQ,3)
+Spec of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
+
+julia> R = OO(X)
+Multivariate Polynomial Ring in x1, x2, x3 over Rational Field
+
+julia> (x1,x2,x3) = gens(R)
+3-element Vector{fmpq_mpoly}:
+ x1
+ x2
+ x3
+
+julia> Y = subscheme(X,ideal(R,[x1*x2]))
+Spec of Quotient of Multivariate Polynomial Ring in x1, x2, x3 over Rational Field by ideal(x1*x2)
+
+julia> defining_ideal(Y)
+ideal(x1*x2)
+```
+"""
+@attr defining_ideal(X::AbsSpec{<:Any, <:MPolyRing}) = ideal(OO(X), [zero(OO(X))])
+defining_ideal(X::AbsSpec{<:Any, <:MPolyQuo}) = modulus(OO(X))
+@attr defining_ideal(X::AbsSpec{<:Any, <:MPolyLocalizedRing}) = ideal(OO(X), [zero(OO(X))])
+defining_ideal(X::AbsSpec{<:Any, <:MPolyQuoLocalizedRing}) = modulus(OO(X))
+
+
+@Markdown.doc """
+    strict_modulus(X::AbsSpec)
+
+This method return the strict modulus of an affine scheme ``X``.
```

Comment:
```suggestion
This method returns the strict modulus of an affine scheme ``X``.
```

Reply:
No clue what a `strict_modulus` is. For me this means that either we need a self explanatory name or the function is internal. Or that we should work more with morphisms.

Reply:
So for now I would leave everything out of the interface that is not rigorously mathematically defined and part of the standard literature.

Reply:
I will commit your suggestion now to fix the typo. Let me remove this entry from the documentation in a separate commit shortly.